book/networking/p2p/network/README.md

```diff
@@ -0,0 +1,285 @@
+- `network_events`: A listener for `NetworkEvent`s sent from the `NetworkManager`, which consist of other "meta" events such as sessions with peers being established or closed.
+- `command_rx`: A listener for `TransactionsCommand`s sent from the `TransactionsHandle`
+- `pending`: A listener for new pending transactions added to the `TransactionPool`
+
+Let's get a view into the transactionst task's operation by walking through the `TransactionManager::poll` method.
+
+### The Operation of the Transactions Task
+
+The `poll` method lays out an order of operations for the transactions task. It begins by draining the `TransactionsManager.network_events`, `TransactionsManager.command_rx`, and `TransactionsManager.transaction_events` streams, in this order.
+Then, it checks on all the current `TransactionsManager.inflight_requests`, which are requests sent by the node to its peers for full transaction objects. After this, it checks on the status of completed `TransactionsManager.pool_imports` events, which are transactions that are being imported into the node's transaction pool. Finally, it drains the new `TransactionsManager.pending_transactions` events from the transaction pool.
+
+Let's go through the handling occurring during each of these steps, in order, starting with the draining of the `TransactionsManager.network_events` stream.
+
+#### Handling `NetworkEvent`s
+
+The `TransactionsManager.network_events` stream is the first to have all of its events processed because it contains events concerning peer sessions opening and closing. This ensures, for example, that new peers are tracked in the `TransactionsManager` before events sent from them are processed.
+
+The events received in this channel are of type `NetworkEvent`:
+
+{{#template ../../../templates/source_and_github.md path_to_root=../../../../ path=crates/net/network/src/manager.rs anchor=enum-NetworkEvent}}
+
+They're handled with the `on_network_event` method, which responds to the two variants of the `NetworkEvent` enum in the following ways:
+
+**`NetworkEvent::SessionClosed`**
+Removes the peer given by `NetworkEvent::SessionClosed.peer_id` from the `TransactionsManager.peers` map.
+
+**`NetworkEvent::SessionEstablished`**
+Begins by inserting a `Peer` into `TransactionsManager.peers` by `peer_id`, which is a struct of the following form:
+
+{{#template ../../../templates/source_and_github.md path_to_root=../../../../ path=crates/net/network/src/transactions.rs anchor=struct-Peer}}
+
+Note that the `Peer` struct contains a field `transactions`, which is an [LRU cache](https://en.wikipedia.org/wiki/Cache_replacement_policies#Least_recently_used_(LRU)) of the transactions this peer is aware of. 
+
+TODO: Is the cache actually _used_ anywhere? (I.e., is `contains` called anywhere on it?) I can't find any such cases...
```

Comment:
Would appreciate thoughts here

Reply:
the implementation is actively used in TransactionManager to keep track of tx hashes a peer has seen.

before POS block announcements were part of the devp2p protocol and it's considered a violation to send a peer a block it has already seen so in POW you still need to keep track of the blocks has seen. ref https://github.com/paradigmxyz/reth/blob/dcd3923d191350386a48a709729731b6dc525e61/crates/net/network/src/manager.rs#L375-L393

the contains is handled in `NetworkState::announce_new_block`